internal/pkg/build/build.go

```diff
@@ -332,8 +353,112 @@ func makeDef(spec string, remote bool) (types.Definition, error) {
 
 	d, err := parser.ParseDefinitionFile(defFile)
 	if err != nil {
-		return types.Definition{}, fmt.Errorf("While parsing definition: %s: %v", spec, err)
+		return types.Definition{}, fmt.Errorf("while parsing definition: %s: %v", spec, err)
+	}
+
+	return d, nil
+}
+
+// MakeAllDefs gets a definition slice from a spec
+func MakeAllDefs(spec string, remote bool) ([]types.Definition, error) {
+	return makeAllDefs(spec, remote)
+}
+
+// makeAllDef gets a definition object from a spec
+func makeAllDefs(spec string, remote bool) ([]types.Definition, error) {
+	if ok, err := uri.IsValid(spec); ok && err == nil {
+		// URI passed as spec
+		d, err := types.NewDefinitionFromURI(spec)
+		return []types.Definition{d}, err
+	}
+
+	// Check if spec is an image/sandbox
+	if _, err := image.Init(spec, false); err == nil {
+		d, err := types.NewDefinitionFromURI("localimage" + "://" + spec)
+		return []types.Definition{d}, err
+	}
+
+	// default to reading file as definition
+	defFile, err := os.Open(spec)
+	if err != nil {
+		return nil, fmt.Errorf("unable to open file %s: %v", spec, err)
+	}
+	defer defFile.Close()
+
+	// must be root to build from a definition
+	if os.Getuid() != 0 && !remote {
+		sylog.Fatalf("You must be the root user to build from a Singularity recipe file")
```

Comment:
Recipe?! Do you mean definition file? :smile: 

This should probably say `recipe`, and to be consistent with are other messages:

```bash
$ singularity build test.sif test.def 
FATAL:   You must be the root user to build from a Singularity recipe file
```

Reply:
@WestleyK are you saying the comment should say recipe or the error should say definition?

Reply:
Err, misread, got excited...

How about this:

```
FATAL:   You must be the root user to build from a definition file
```